Fix build against OpenSSL 3.0

I noticed that @schopin-pro recently [patched](http://launchpadlibrarian.net/572913938/sscg_3.0.1-2_3.0.1-2ubuntu1.diff.gz) sscg to fix a [build failure](https://launchpadlibrarian.net/572403390/buildlog_ubuntu-jammy-amd64.sscg_3.0.1-2build1_BUILDING.txt.gz) against OpenSSL 3.0 in Ubuntu:
```
../src/dhparams.c:237:8: error: implicit declaration of function ‘FIPS_mode’ [-Werror=implicit-function-declaration]
  237 |   if (!FIPS_mode ())
      |        ^~~~~~~~~
cc1: some warnings being treated as errors
```

At first sight that patch looks quite reasonable for upstream?

--- a/src/dhparams.c
+++ b/src/dhparams.c
@@ -234,7 +234,11 @@ is_valid_named_group (const char *group_name)
     }
 
   /* Check non-FIPS groups */
+#if OPENSSL_VERSION_NUMBER < 0x30000000L
   if (!FIPS_mode ())
+#else
+  if (!EVP_default_properties_is_fips_enabled(NULL))
+#endif
     {
       i = 0;
       while (dh_nonfips_groups[i])

